# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformFontList.h
# Commit: a01eaeed432a
# Full Hash: a01eaeed432ab996a3f5275229be89814e92b95a
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1708768
# File Overlap Count: 2
# Description:
#   Bug 1708768 - Preliminary cleanup of gfxPlatformFontList creation/initialization code; no functional change. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114863
# ==============================================================================

diff -r 39c32384d7a9 -r a01eaeed432a gfx/thebes/gfxPlatformFontList.h
--- a/gfx/thebes/gfxPlatformFontList.h	Tue May 18 10:47:43 2021 +0000
+++ b/gfx/thebes/gfxPlatformFontList.h	Tue May 18 11:10:42 2021 +0000
@@ -171,13 +171,13 @@
 
   static gfxPlatformFontList* PlatformFontList() { return sPlatformFontList; }
 
-  static nsresult Init() {
-    NS_ASSERTION(!sPlatformFontList, "What's this doing here?");
-    gfxPlatform::GetPlatform()->CreatePlatformFontList();
-    if (!sPlatformFontList) {
-      return NS_ERROR_OUT_OF_MEMORY;
+  static bool Initialize(gfxPlatformFontList* aList) {
+    sPlatformFontList = aList;
+    if (aList->InitFontList()) {
+      return true;
     }
-    return NS_OK;
+    Shutdown();
+    return false;
   }
 
   static void Shutdown() {
@@ -187,8 +187,8 @@
 
   virtual ~gfxPlatformFontList();
 
-  // initialize font lists
-  nsresult InitFontList();
+  // Initialize font lists; return true on success, false if something fails.
+  bool InitFontList();
 
   void FontListChanged();
 